### ISSUE TITLE
Focal loss forward

### ISSUE THREAD

**Issue by Krual-T** [owj0421/OutfitTransformer#18]
Your loss function uses binary_cross_entropy_with_logits when calculating. However, this function should accept values that have not been activated. In outfit_transformer.py, the activation layer of predict_ffn is defined as sigmoid, which is already activated probability. Therefore, what you should use in the loss is binary_cross_entropy instead of binary_cross_entropy_with_logits.

**Comment by pbtaffi** [comment]
> > Your loss function uses binary_cross_entropy_with_logits when calculating. However, this function should accept values that have not been activated. In outfit_transformer.py, the activation layer of predict_ffn is defined as sigmoid, which is already activated probability. 

good catch @Krual-T.


> gdown --id 1mzNqGBmd8UjVJjKwVa5GdGYHKutZKSSi -O checkpoints.zip

Hello @owj0421   
1. the checkpoints that you provided in readme - are they reliable or created using this problematic loss function. 
2. do you have any more checkpoints for clip or the original model.
3. the performance numbers that you provided in readme - are they using this checkpoint ?

Thanks.

**Comment by Krual-T** [comment]
这是来自QQ邮箱的假期自动回复邮件。您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by bigohofone** [comment]
Thanks for the information. I’ll change it to the correct loss function ASAP.
